FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -4468,4 +4468,67 @@ - (void)testGetSkipsPersistenceCacheWhenOnline {
   WAIT_FOR(done);
 }
 
+- (void)testGetTriggersListener {
+  FIRDatabase* db = [self databaseForURL:self.databaseURL name:@"fir-multi-shards"];
+
+  FIRDatabaseReference* ref = [db referenceWithPath:@"james"];
+  FIRDatabaseReference* children = [ref child:@"children"];
+
+  __block BOOL done = NO;
+
+  [children removeValueWithCompletionBlock:^(NSError* _Nullable error,
+                                             FIRDatabaseReference* _Nonnull ref) {
+    XCTAssertNil(error);
+    done = YES;
+  }];
+
+  WAIT_FOR(done);
+  done = NO;
+
+  [[children childByAutoId] setValue:@{@"name" : @1}
+                 withCompletionBlock:^(NSError* error, FIRDatabaseReference* ref) {
+                   XCTAssertNil(error);
+                   done = YES;
+                 }];
+
+  WAIT_FOR(done);
+  done = NO;
+
+  [[children childByAutoId] setValue:@{@"name" : @2}
+                 withCompletionBlock:^(NSError* error, FIRDatabaseReference* ref) {
+                   XCTAssertNil(error);
+                   done = YES;
+                 }];
+
+  WAIT_FOR(done);
+  done = NO;
+
+  __block BOOL nextDone = NO;
+
+  [children observeEventType:FIRDataEventTypeValue
+                   withBlock:^(FIRDataSnapshot* snapshot) {
+                     if (done == YES) {
+                       XCTAssertTrue(snapshot.childrenCount == 2U);
+                       nextDone = YES;
+                       return;
+                     }
+                     XCTAssertTrue(snapshot.childrenCount == 2U);
+                     done = YES;
+                   }];
+
+  WAIT_FOR(done);
```

Comment:
The original repo also does not wait for "done". It runs both operation in parallel.